### PR TITLE
PoC for explicitization of dynamic vars

### PR DESCRIPTION
**Pull request by escherize**

### CODE REVIEW

File: src/metabase/api/card.clj [hunk 1]
```diff
@@ -516,14 +516,14 @@ saved later when it is ready."
   transaction and work in the `:card-create` event cannot proceed because the cards would not be visible outside of
   the transaction yet. If you pass true here it is important to call the event after the cards are successfully
   created."
-  ([card] (create-card! card false))
-  ([{:keys [dataset_query result_metadata dataset parameters parameter_mappings], :as card-data} delay-event?]
+  ([creator card] (create-card! creator card false))
+  ([creator {:keys [dataset_query result_metadata dataset parameters parameter_mappings], :as card-data} delay-event?]
```
Comment (tsmacdonald) [inline src/metabase/api/card.clj:520]: feels like the card-data should be either first or last, not in the middle of the params

Comment (escherize) [reply src/metabase/api/card.clj:520]: That's fair I'll swap the arg order to make it come first.

**Review by noahmoss — CHANGES_REQUESTED** [review]
-

**Review by noahmoss — APPROVED** [review]
Makes sense to me. I think cases like this where a user is required are the ones where we should be more explicit (vs cases where the user isn't always required, like audit logging)

**Comment by noahmoss** [comment]
(clicked the wrong button the first time, ignore the request for changes)

**Review by calherries — APPROVED** [review]

**Pull request merged**